Minor xmp  fixes

chkuendig opened the pull request:
- ~Don't write xmp sidecar for adjusted versions (some metadata like orientation only makes sense when applying to l the original file)~
- Some images (screenshots?) have a different 'adjustmentSimpleDataEnc' (plist instead oc compressed JSON) field. This led to a crash
- One comment had the wrong indents

AndreyNikiforov commented:
> * Some images (screenshots?) have a different 'adjustmentSimpleDataEnc' (plist instead oc compressed JSON) field. This led to a crash

For `filenameEnc` field we have to evaluate `type` subfield to get correct encoding. Does `adjustmentSimpleDataEnc` have `type` differentiating encoding too? We can use it to support encoding explicitly and skip unsupported types - probably good to do that for all enc fields to avoid breakages when Apple introduce new encoding.

chkuendig commented on src/icloudpd/base.py:
```diff
@@ -970,7 +970,7 @@ def download_photo_(counter: Counter, photo: PhotoAsset) -> bool:
                             logger.info("Downloaded %s", truncated_path)
 
-                if xmp_sidecar:
+                if xmp_sidecar and download_size != AssetVersionSize.ADJUSTED:
```
Yeah, it might make sense to think this through better. Maybe only download the xmp for original? Generally since the XMP is based on the asset record, it should only be created once for a specific foto.

chkuendig commented:
> For filenameEnc field we have to evaluate type subfield to get correct encoding. Does adjustmentSimpleDataEnc have type differentiating encoding too? We can use it to support encoding explicitly and skip unsupported types - probably good to do that for all enc fields to avoid breakages when Apple introduce new encoding.

Unfortunately for all fields ending in `Enc` the type subfield just says `ENCRYPTED_BYTES` which afaik can be any of the following three (maybe there's more):
- Base64 encoded bytes for a unicode string
- Base64 encoded bytes of a binary plist dict
- Base64 encoded bytes of zlib compressed JSON

The Plist is easy to detect (it has specific header bytes), the other two can only be detected by decoding it (and in theory you could probably have a zlib compressed JSON where all bytes are valid unicode?).

chkuendig commented:
@AndreyNikiforov I removed the limitation on the `AssetVersionSize` so this can be unblocked. Maybe long term this is something to revisit, but it doesn't cause any issues as is I think (unlike the crash that's also fixed in this PR).

chkuendig commented:
PS: Please hold, I found one more bug.

AndreyNikiforov commented:
@chkuendig can you pls add test(s). Sorry, I should have mentioned that sooner.

AndreyNikiforov commented:
@chkuendig are you okay with PR or you have something else to finish? If all good, change PR "draft" status to normal - I cannot merge drafts.

chkuendig commented:
I added tests but forgot to push. All good now, I also retested it on my library without issues.
